lib/CPlayerState.h

```diff
@@ -39,6 +39,7 @@ struct DLL_LINKAGE PlayerState : public CBonusSystemNode, public Player
 	std::vector<QuestInfo> quests; //store info about all received quests
 	std::vector<Bonus> battleBonuses; //additional bonuses to be added during battle with neutrals
 
+	bool cheated;
```

Comment:
Perhaps remove `PlayerMessageProcessor::cheaters` field since it now duplicates this field?